components/infobox/extensions/commons/metadata_generator.lua

```diff
@@ -0,0 +1,172 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:MetadataGenerator
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local String = require('Module:String')
+local Localisation = require('Module:Localisation')
+local Template = require('Module:Template')
+local Games = require('Module:Games')
+local Variables = require('Module:Variables')
+local StringUtils = require('Module:StringUtils')
+local Class = require('Module:Class')
+
+local MetadataGenerator = {}
+
+function MetadataGenerator.tournament(args)
+	if String.isEmpty(args.publisherdescription) then
+		error('You must provide the publisherdescription param!')
+	end
+
+	local output
+	local frame = mw.getCurrentFrame()
+
+	local name = not String.isEmpty(args.name) and (args.name):gsub('&nbsp;', ' ') or mw.title.getCurrentTitle()
+
+	local type = args.type
+	local locality = Localisation.getLocalisation({displayNoError = true}, args.country)
+
+	local organizers = {
+		args['organizer-name'] or args.organizer,
+		args['organizer2-name'] or args.organizer2,
+		args['organizer3-name'] or args.organizer3,
+		args['organizer4-name'] or args.organizer4,
+		args['organizer5-name'] or args.organizer5
+	}
+
+	local tier = args.liquipediatier and Template.safeExpand(frame, 'TierDisplay', {args.liquipediatier}) or nil
+
+	if tier then
+		tier = (tonumber(
+			Template.safeExpand(frame, 'TierDisplay/number', {args.liquipediatier})
+		) or 0) > 4 and tier:lower() or tier
+	else
+		tier = 'Unknown Tier'
+	end
+
+	local tierType = (tier == 'qualifier' or tier == 'showmatch') and tier or 'tournament'
+	local publisher = Variables.varDefault(args.publisherdescription, '')
+	local date, tense = MetadataGenerator.getDate(args.edate or args.date, args.sdate)
+
+	local teams = args.team_number
+	local players = args.player_number
+
+	local game = (not String.isEmpty(args.game) and args.game ~= 'csgo') and Games.abbr[args.game]
```

Comment:
should be generalized i think

Reply:
i started the review when the comment wasn't there yet, then got distracted by something work related and continued afterwards, only saw the comment afterwards^^

Reply:
Done